src/Cardano/Shell/Constants/Types.hs

```diff
@@ -65,11 +65,20 @@ data Core = Core
       -- ^ Versioning for values in node's DB.
     } deriving (Eq, Show)
 
-data Genesis = Genesis { geInternal :: !Bool
-                       , geSpec     :: !Spec
-                       , geSrc      :: !FilePath
-                       , geFileHash :: !Text
-                       } deriving (Eq, Show)
+-- | The genesis section.
+-- For now, we only store the path to the genesis file(s) and their hash.
+-- The rest is in the hands of the modules/features that need to use it.
+-- The info flow is:
+-- __genesis config ---> genesis file__
+-- And separately:
+-- __genesis file ---> runtime config ---> running node__
+-- __static config ---> ...__
+--
+data Genesis = Genesis
+    { geSrc             :: !FilePath
+    , geGenesisHash     :: !Text
+    , gePrevBlockHash   :: !Text
+    } deriving (Eq, Show)
 
```

Comment:
So my understanding is that we also need the `Spec` if we want to launch the node with a custom configuration. Do we really want to remove this? What is the alternative?

Reply:
That is something I was bothered with as well.
For example, we don't have the `k` parameter unless we have a "custom" configuration. And that "custom" configuration contains some parameters I'm not sure are required.
What I can say for now is that it seems we don't need them right away, and thus I simply removed them. I want to see what ends up missing rather than have some additional params which serve no use at all.
The types remain and if you do end up requiring specific parameters, I will return them easily.